ha/setup/cluster_validator/cluster_status.py

```diff
@@ -0,0 +1,275 @@
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+Module to provide standalone interface to cluster status.
+"""
+
+import re
+from typing import Callable, List, NamedTuple, Union, Optional
+from xml.etree import ElementTree
+from xml.etree.ElementTree import Element
+from ha.core.error import SetupError, HA_CLUSTER_CONFIG_ERROR
+from ha.execute import SimpleCommand
+
+
+class ValidationStatusError(SetupError):
+    """Cluster status format is invalid."""
+    def __init__(self, desc=None):
+        _desc = "Error when parsing cluster status" if desc is None else desc
+        _message_id = HA_CLUSTER_CONFIG_ERROR
+        _rc = 1
+        super().__init__(rc=_rc, desc=_desc, message_id=_message_id)
+
+
+ClusterSummary = NamedTuple("ClusterSummary", [
+    ("maintenance_mode", bool),
+    ("quorum", bool),
+    ("stonith_enabled", bool),
+    ("num_nodes", int),
+    ("num_resources_configured", int),
+    ("num_resources_disabled", int)
+])
+
+ClusterNode = NamedTuple("ClusterNode", [
+    ("name", str),
+    ("online", bool),
+    ("standby", bool),
+    ("maintenance", bool),
+    ("unclean", bool),
+    ("resources_running", int),
+])
+
+ClusterResource = NamedTuple("ClusterResource", [
+    ("name", str),
+    ("clone", bool),
+    ("resource_agent", str),
+    ("role", str),  # Enum may be needed here
+    ("enabled", bool),
+    ("failed", bool),
+    ("managed", bool),
+    ("failure_ignored", bool),
+    ("location", List[str]),
+    ("group", str),
+])
+
+# NOTE: Non-empty group attribute indicates that it is not a resource, but a cloned group
+ClusterCloneResource = NamedTuple("ClusterCloneResource", [
+    ("name", str),
+    ("managed", bool),
+    ("unique", bool),
+    ("failed", bool),
+    ("failure_ignored", bool),
+    ("copies", List[ClusterResource]),
+    ("group", str),
+])
+
+
+class ClusterStatusPcs():
+    """Implementation of interface for pcs status --xml command."""
+
+    def __init__(self, executor: Callable[[str], tuple] = None):
+        """Call `pcs status` and init XML object for further parsing"""
+        if not executor:
+            executor = SimpleCommand().run_cmd
+        output, _, _ = executor("pcs status --full xml")
+        self.tree = ElementTree.fromstring(output)
+
+    @staticmethod
+    def _to_bool(value: str) -> bool:
+        return value == "true"
+
+    def get_nodes(self) -> List[ClusterNode]:
+        """Return list of nodes."""
+        node_elements = self.tree.findall("./nodes/node")
+        nodes = []
+        try:
+            b = self._to_bool
+            for n in node_elements:
+                nodes.append(ClusterNode(
+                    name=n.attrib["name"],
```

Comment:
There are lot of string literals used everywhere.
Please group these and put then in different classes in const.py. Example -
class ClusterStatusAttributes:
       ONLINE = "online"
       STANDBY = "standby"

Also, I think that it would be better to use them in such a way that case can be ignored.